### ISSUE TITLE
[Keyboard Navigation - Microbit Menu Controls] Focus indicator is navigating to Seperator after 'Math' control Via Tab key.

### ISSUE THREAD

**Issue by v-mesala** [Microsoft/pxt#2610]
**User Experience:**
Users with Visual disabilities might get confuse if focus is navigating to non interactive elements. Like in the scenario Focus is navigating to separator and reads information as one of list item.

**Test Environment**
OS: RS2 Version 1703(OS Build 15063.2)
Platform: Edge.
Screen reader: Narrator

**Repro Steps**
1-Navigate to url:- https://makecode.com/
2-Navigate to the Micro bit section element and select Code control given on it.
3-Navigate to the menu controls listed near the Simulator control.
4-Navigate to the separator and verify the issue.

**Actual Result**
Focus indicator is navigating to non interactive element i.e Separator line and narrator reads as 1 of 12.

**Expected Result**
Focus indicator should not navigate to separator line which is non interactive element.

**MAS Reference**
https://microsoft.sharepoint.com/teams/msenable/Documents/LCA/MAS%20-%20July%201,%202017%20Release/MAS%202.4.3%20%E2%80%93%20Focus%20Order%20.docx?d=w0de7fbe1ad7e48e5bcbb8d986691e2b9&csf=1

**Suggested fix:**
In all the applicable forms we need to check below-  
1.Check if 'tabindex' is used 
2.If 'tabindex' is used, check that the tab order specified by the 'tabindex' attributes follows relationships in the content. 
tabindex property is applicable to all visual elements on screen.  
<a href="/articles/" tabindex="2">Articles</a> 
<a href="/resources/" tabindex="3">Resources</a> 
 
Tips : 
'tabindex' should only be used in cases where: 
•The default tab order is not ideal, AND 
•The tab order cannot be changed by rearranging items in the content and/or by altering the style sheet to reflect the best visual arrangement. 
 
**Key References for Fix:** 
http://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20160317/G59 
https://www.w3.org/WAI/GL/wiki/Creating_Logical_Tab_Order_with_the_Tabindex_Attribute 

**Refer Attachment:**
[MAS 2.4.3_Focus is navigating to seperator.zip](https://github.com/Microsoft/pxt/files/1190524/MAS.2.4.3_Focus.is.navigating.to.seperator.zip)

**Comment by veler** [comment]
This is part of Blocky, a third-part component.

**Comment by abchatra** [comment]
We have got exception for Blockly bugs from Julia Liuson (CVP). Closing these as Won't Fix. Following is the reasoning.

----------------------

MakeCode can be used in either Text editor or Block editor.

Blocks editor is inherently visual and is based on google blockly. Usage requires drag and drop functionality and we don’t have a equivalence in MakeCode for this w.r.t accessibility. 
We have evaluated couple of research projects which tries to find a equivalence to block based coding for accessibility needs: 
•	Project Torino (Requires students to buy physical blocks)
•	Accessible blockly (This is text based coding without syntax errors)
Both these are drastic departure from the blockly and provides a complete new experience. These are still in research and we intend not to use them in MakeCode at this point of time.

As workaround, the text editor in MakeCode is the way we see students with the need of accessibility using MakeCode.
